src/gt4py/next/embedded/common.py

```diff
@@ -97,14 +97,57 @@ def _absolute_sub_domain(
     return common.Domain(*named_ranges)
 
 
-def intersect_domains(*domains: common.Domain) -> common.Domain:
+def domain_intersection(
+    *domains: common.Domain,
+) -> common.Domain:
+    """
+    Return the intersection of the given domains.
+
+    Example:
+    >>> I = common.Dimension("I")
+    >>> domain_intersection(
+    ...     common.domain({I: (0, 5)}), common.domain({I: (1, 3)})
+    ... )  # doctest: +ELLIPSIS
+    Domain(dims=(Dimension(value='I', ...), ranges=(UnitRange(1, 3),))
+    """
     return functools.reduce(
         operator.and_,
         domains,
         common.Domain(dims=tuple(), ranges=tuple()),
     )
 
 
+def intersect_domains(
```

Comment:
- Changed the name.
- In generic context allowing ignore_dims to be optional makes sense to me. Maybe I could change the API to only accept tuples and default to empty tuple. What do you think?

Reply:
I still don't like that much that `ignore_dims` is optional but it's ok. `ignore_dims` accepting only tuples makes sense to me, but it's not strictly needed and it's orthogonal to my original complaint because it'll still have a default value.